book-1-orientation/chapters/DATA_STRUCTURES_DICTIONARY.md

```diff
@@ -41,157 +41,230 @@ foreach(KeyValuePair<string, int> toy in toysSold)
 
-> **Helpful Links:** [ContainsKey](https://msdn.microsoft.com/en-us/library/kw5aaea4(v=vs.110).aspx), [Add](https://msdn.microsoft.com/en-us/library/k7z0zy8k(v=vs.110).aspx)
 
 ```cs
+// Make a new list
+List<Dictionary<string, string>> dictionaryOfWords = new List<Dictionary<string, string>>();
+
 /*
-    Define a new Dictionary to hold the aggregated purchase information.
-    - The key should be a string that is the full company name.
-    - The value will be the total valuation of each stock
+    You want to track the following about each word:
+        word, definition, part of speech, example sentence
 
+    Example of one dictionary in the list:
+    {
+        "word": "excited",
+        "definition": "having, showing, or characterized by a heightened state of energy, enthusiasm, eagerness",
+        "part of speech": "adjective",
+        "example sentence": "I am excited to learn C#!"
+    }
+*/
 
-    From the three purchases above, one of the entries
-    in this new dictionary will be...
-        {"General Electric", 1217.53}
+// Create dictionary to represent a few word
+Dictionary<string, string> excitedWord = new Dictionary<string, string>();
+
+// Add each of the 4 bits of data about the word to the Dictionary
+excitedWord.Add();
+
+// Add Dictionary to your `dictionaryOfWords` list
+
+
+// create another Dictionary and add that to the list
 
-    Replace the questions marks below with the correct types.
-*/
-Dictionary<?, ?> stockReport = new Dictionary<?, ?>();
 
 /*
-   Iterate over the purchases and record the valuation
-   for each stock.
+    Iterate your list of dictionaries and output the data
+
+    Example output for one word in the list of dictionaries:
+        word: excited
+        definition: having, showing, or characterized by a heightened state of energy, enthusiasm, eagerness
+        part of speech: adjective
+        example sentence: I am excited to learn C#!
 */
-foreach (Dictionary<string, double> purchase in purchases)
+
+// Iterate the List of Dictionaries
+foreach ()
 {
-    foreach (KeyValuePair<string, double> stock in purchase)
+    // Iterate the KeyValuePairs of the Dictionary
+    foreach ()
     {
-        // Does the full company name key already exist in the `stockReport`?
-
-        // If it does, update the total valuation
-
-        /*
-            If not, add the new key and set its value.
-            You have the value of "GE", so how can you look
-            the value of "GE" in the `stocks` dictionary
-            to get the value of "General Electric"?
-        */
+        Console.WriteLine($"{wordData.Key}: {wordData.Value}");
     }
 }
 ```
 
-Now that the report dictionary is populated, display the final results.
+## Practice: English Idioms
```

Comment:
There are a lot of exercises here now. I think I would cut this one.

Reply:
There are only three exercises, one challenge and one advanced challenge. Pretty standard set for the course. I'm going to leave it in for now, but if we find over the next couple cohorts that it adds too much time, we can rip it out.  Since this is a very early chapter, I don't believe there's such a thing as **too much** practice.